book_source/source/2darrays.md

```diff
@@ -0,0 +1,237 @@
+# <i class="fas fa-book fa-fw"></i> 2DArrays
+
+## 2DArray Fundamentals
+
+An 2D array is very similar to an 1D array, or just an array as we have learned, only that instead of storing primiative types, it stores another array. In other words, a 2D array is an array where each element contains another array. Commonly this is represented/used when dealing with images, graphs, tables, and various other 2D representations.
+
+Much like how you would access a specific element in an array using square brackets '[]', for a 2D array, you need to add one more pair of brackets to access an element in a 2D array. For example, if you wanted to create a 2D array storing integers, an example syntax could be int[][] (read as a 2D array of ints or 2D int array). Although the specific lengths of the arrays can vary, these arrays still have a fixed length once created. For ease of representation, we typically repersent a 2D array with rows being the inner arrays and columns being the array of arrays. For example:
```

Comment:
While we could mention jagged 2D arrays, I don't think it's worth getting into -- they're not really covered in depth in 12x and we'll never ask them to deal with them.